addons/stock/models/stock_move.py

```diff
@@ -358,17 +358,19 @@ def _compute_quantity(self):
 
     def _set_quantity(self):
         def _process_decrease(move, quantity):
+            mls_to_unlink = self.env['stock.move.line']
```

Comment:
```suggestion
            mls_to_unlink_ids = set()
```
It's usually faster to register the ids then browse & do whatever you need to instead of piping records.

Reply:
Nice to know thanks !